numpy/core/src/umath/loops.c.src

```diff
@@ -1617,6 +1631,42 @@ NPY_NO_EXPORT void
 }
 /**end repeat1**/
 
+NPY_NO_EXPORT void
+@TYPE@_max_abs(char **args, npy_intp *dimensions, npy_intp *steps, void* NPY_UNUSED(func))
+{
+#if @uint_matched@
+    /* Everything except signbit set, 0x7fffffffffffffff */
+    const @type@ mask0 = -0.0@c@;
+    const @uint_type@ mask = ~(*((@uint_type@*)&mask0));
+    if (IS_BINARY_REDUCE) {
+        if (!run_unary_reduce_simd_max_abs_@TYPE@(args, dimensions, steps)) {
+            BINARY_REDUCE_LOOP(@type@) {
+                const @uint_type@ uin2 = (*(@uint_type@ *)ip2) & mask;
+                const @type@ in2 = *((@type@ *)&uin2);
+                io1 = (io1 >= in2 || npy_isnan(io1)) ? io1 : in2;
+            }
+        }
+    }
+    else {
+        BINARY_LOOP {
+            const @uint_type@ uin1 = (*(@uint_type@ *)ip1) & mask;
+            const @uint_type@ uin2 = (*(@uint_type@ *)ip2) & mask;
+            const @type@ in1 = *((@type@ *)&uin1);
+            const @type@ in2 = *((@type@ *)&uin2);
+            *((@uint_type@ *)op1) = (in1 >= in2 || npy_isnan(in1)) ? in1 : in2;
+         }
+    }
+#else
+    BINARY_LOOP {
+        @type@ in1 = *(@type@ *)ip1;
+        @type@ in2 = *(@type@ *)ip2;
+        in1 = (in1 >= 0) ? in1 : -in1;
+        in2 = (in2 >= 0) ? in2 : -in2;
+        *((@type@ *)op1) = (in1 >= in2 || npy_isnan(in1)) ? in1 : in2;
+     }
```

Comment:
this ideally have a BINARY_REDUCE_LOOP specialization as it is faster for reductions